superset/assets/src/explore/exploreUtils.js

```diff
@@ -32,6 +32,12 @@ let requestCounter = 0;
 function getHostName(allowDomainSharding = false) {
   let currentIndex = 0;
   if (allowDomainSharding) {
+    // if domain sharding is enabled, skip main domain for fetching chart API
+    // leave main domain free for other calls like fav star, save change, etc.
+    // to make dashboard be responsive when it's loading large number of charts
+    if (requestCounter % availableDomains.length === 0) {
+      requestCounter += 1;
+    }
```

Comment:
minor nit: since you're duplicating the `requestCounter % availableDomains.length` logic from below, might it be simpler to do:

```js
if (allowDomainSharding) {
  currentIndex = requestCounter % availableDomains.length;
  requestCounter += 1;

  // if domain sharding is enabled, skip main domain for fetching chart API
  // leave main domain free for other calls like fav star, save change, etc.
  // to make dashboard be responsive when it's loading large number of charts
  if (currentIndex === 0) {
    currentIndex += 1;
    requestCounter += 1;
  }
}
```

Reply:
changes are added.